### PR TITLE
Compiler: Add amp-carousel-0.1 to the builder map

### PR DESCRIPTION
**Pull request by samouri**
**summary**
A necessary step for ssring `amp-carousel` is adding it to the builder map.

### CODE REVIEW

File: src/compiler/builders.js [hunk 1]
```diff
@@ -10,6 +11,7 @@ const versionedBuilderMap = {
   },
   '0.1': {
     'amp-fit-text': ampFitTextClassic,
+    'amp-carousel': ampCarouselClassic1,
```
Comment (rcebulko) [inline src/compiler/builders.js:14]: `ampCarouselClassic1` thanks I hate it 😆

**Review by rcebulko — APPROVED** [review]

File: extensions/amp-carousel/0.1/build-dom.js [hunk 2]
```diff
@@ -317,13 +319,16 @@ export function getPrevButtonTitle(element, options = {}) {
  * - Slides    : "Next item in carousel (X of Y)"
  *
  * @param {*} element
- * @param {{prefix: string, index: string, total:string}} param1
+ * @param {{prefix: string, index?: string, total?:string}} param1
```
Comment (rcebulko) [inline extensions/amp-carousel/0.1/build-dom.js:322]:
```suggestion
 * @param {{prefix: string, index?: string, total?: string}} opts
```

File: extensions/amp-carousel/0.1/build-dom.js [hunk 3]
```diff
@@ -317,13 +319,16 @@ export function getPrevButtonTitle(element, options = {}) {
...
  * @return {string}
  */
 function getButtonTitle(element, {index, prefix, total}) {
   if (isScrollable(element)) {
     return prefix;
   }
+  // Slides carousel should always have index/total provided.
+  devAssert(index);
+  devAssert(total);
```
Comment (rcebulko) [inline extensions/amp-carousel/0.1/build-dom.js:331]: What is this meant to check? That the strings are non-empty?

Comment (samouri) [reply extensions/amp-carousel/0.1/build-dom.js:331]: Yes

File: extensions/amp-carousel/0.1/build-dom.js (hunk 2 above)
Comment (samouri) [reply extensions/amp-carousel/0.1/build-dom.js]: updated

File: extensions/amp-carousel/0.1/build-dom.js [hunk 4]
```diff
@@ -287,9 +288,9 @@ export function buildDom(element) {
 /**
  * @param {Element} element
  * @return {string} The default title to use for the next button.
- * @param {{index?: string, total?: string}} options - The default title to use for the previous button.
+ * @param {{index: string, total: string}} options - The default title to use for the previous button.
  */
-export function getNextButtonTitle(element, options = {}) {
+export function getNextButtonTitle(element, options) {
```
Comment (rcebulko) [inline extensions/amp-carousel/0.1/build-dom.js:293]: nit: If you end up with any other changes, may as well destructure the args here and `getPrevButtonTitle` like you do in `getButtonTitle`. If not, w/e
```suggestion
export function getNextButtonTitle(element, {index, total}) {
```

**Review by rcebulko — APPROVED** [review]

**Pull request merged**